### PR TITLE
[caffe2] Add Moments to math

### PR DESCRIPTION
**Pull request by xiaomengy**

### CODE REVIEW

File: caffe2/utils/math.h [hunk 1]
```diff
@@ -207,6 +207,18 @@ void Broadcast(
     T* Y,
     Context* context);
 
+template <typename T, class Context>
+void Moments(
```
Comment (yinghai) [inline caffe2/utils/math.h]: What's moments? In maths, it usually comes with n-th order, right?

Comment (BIT-silence) [reply caffe2/utils/math.h]: It just comes from TensorFlow's moments which computes the mean and variance of a given tensor. Maybe we can consider a better name.
https://www.tensorflow.org/api_docs/python/tf/nn/moments

Comment (yinghai) [reply caffe2/utils/math.h]: I see. Mean and variance are first and second moments. Makes sense.

**Review by houseroad — APPROVED** [review]
LGTM, thanks!

**Pull request merged**